### ISSUE TITLE
google_secret_manager_secret in for_each loop returns Error 400: Secret must be provided.

### ISSUE THREAD

**Issue by nikkatalnikov** [hashicorp/terraform-provider-google#11395]
### Affected Resource(s)

* google_secret_manager_secret

Trying to create multiple secrets from the JSON collection:

```terraform
resource "google_secret_manager_secret" "secret-general" {
  project       = "modern-project-123"
  for_each      = local.general_secrets
  secret_id     = each.key

  replication {
    automatic = false
  }
}

resource "google_secret_manager_secret_version" "secret-version-general" {
  for_each      = google_secret_manager_secret.secret-general
  secret        = each.value.name
  secret_data   = local.general_secrets[each.key]
}

```

And getting an error

```
│ Error: Error creating Secret: googleapi: Error 400: Secret must be provided.
│ 
│   with google_secret_manager_secret.secret-general["API_KEY"],
│   on 02-general-secrets.tf line 1, in resource "google_secret_manager_secret" "secret-general":
│    1: resource "google_secret_manager_secret" "secret-general" {
│ 
```

Env:

Terraform v1.1.2 on darwin_amd64
provider registry.terraform.io/hashicorp/google v4.15.0

Is that a bug or what am i doing wrong? Thank you!

b/301412033

**Comment by nikkatalnikov** [comment]
I found that it works with 

```
  replication {
    automatic = true
  }
```

but why?

**Comment by nikkatalnikov** [comment]
@rileykarson 
I can confirm it works if replication is set like in your example

**Comment by nikkatalnikov** [comment]
probably it worth to fix an error message to resolve the issue?

**Comment by fclabs** [comment]
I am having a similar issue with

terraform v1.2.4
provider version = "4.27.0"

Debug shows an error in the API call, where an empty body is shown.

``` 
---[ REQUEST ]---------------------------------------
POST /v1/projects/wordpress-in-gks/secrets/gke-cluster-master-auth-client-certificate:addVersion?alt=json HTTP/1.1
Host: secretmanager.googleapis.com
User-Agent: Terraform/1.2.4 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Length: 3
Content-Type: application/json
Accept-Encoding: gzip

{}

-----------------------------------------------------
```

And the response:
```
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Thu, 07 Jul 2022 17:30:27 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "error": {
    "code": 400,
    "message": "Field [payload] is required.",
    "status": "INVALID_ARGUMENT"
  }
}

-----------------------------------------------------
```

**Comment by slevenick** [comment]
Fixed in 5.0+
